File: backend/src/main/java/br/ufpr/sept/so2/modules/solicitacoes/api/dto/SolicitacaoResponse.java
package br.ufpr.sept.so2.modules.solicitacoes.api.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.OffsetDateTime;
import java.util.List;
import java.util.Map;
import java.util.UUID;

public record SolicitacaoResponse(
        UUID id,
        String protocolo,
        String tipoCodigo,
        String tipoNome,
        int tipoVersao,
        String estado,
        Map<String, Object> payload,
        Map<String, Object> formSchema,
        OffsetDateTime prazoEm,
        boolean prazoVencido,
        String sla,
        OffsetDateTime createdAt,
        OffsetDateTime updatedAt,
        List<SolicitacaoEventoResponse> eventos,
        @JsonProperty("_links") Map<String, String> links
) {
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/solicitacoes/api/dto/SolicitacaoResponse.kt
package br.ufpr.sept.so2.modules.solicitacoes.api.dto

import com.fasterxml.jackson.annotation.JsonProperty
import java.time.OffsetDateTime
import java.util.UUID

data class SolicitacaoResponse(
    val id: UUID,
    val protocolo: String,
    val tipoCodigo: String,
    val tipoNome: String,
    val tipoVersao: Int,
    val estado: String,
    val payload: Map<String, Any?>,
    val formSchema: Map<String, Any?>?,
    val prazoEm: OffsetDateTime?,
    val prazoVencido: Boolean,
    val sla: String,
    val createdAt: OffsetDateTime?,
    val updatedAt: OffsetDateTime?,
    val eventos: List<SolicitacaoEventoResponse>,
    @get:JsonProperty("_links")
    val links: Map<String, String>,
)
